src/win/util.c

```diff
@@ -1153,3 +1155,57 @@ int uv_getrusage(uv_rusage_t *uv_rusage) {
 
   return 0;
 }
+
+
+int uv_os_homedir(char* buffer, size_t* size) {
+  wchar_t* path;
+  HRESULT hr;
+  char* buf;
+  size_t utf8_len;
+  int r;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  hr = SHGetKnownFolderPath(&FOLDERID_Profile, 0, NULL, &path);
+
+  if (hr != S_OK)
+    return uv_translate_sys_error(WSAGetLastError());
+
+  /* Determine the number of bytes needed to store the path in UTF-8 */
+  utf8_len = WideCharToMultiByte(CP_UTF8, 0, path, -1, NULL, 0, NULL, NULL);
+
+  if (utf8_len == 0) {
+    CoTaskMemFree(path);
+    return uv_translate_sys_error(WSAGetLastError());
+  }
+
+  buf = malloc(utf8_len);
+
+  if (buf == NULL) {
+    CoTaskMemFree(path);
+    return -ENOMEM;
+  }
+
+  /* path is null terminated, so -1 can be passed as the 4th argument */
+  r = WideCharToMultiByte(CP_UTF8, 0, path, -1, buf, utf8_len, NULL, NULL);
+  CoTaskMemFree(path);
+
+  if (r == 0) {
+    free(buf);
+    return uv_translate_sys_error(WSAGetLastError());
+  }
+
+  assert((unsigned int) r == utf8_len);
+
+  /* Copy buf to buffer, truncating if necessary */
+  if (utf8_len > *size)
+    utf8_len = *size;
+
+  memcpy(buffer, buf, utf8_len);
+  free(buf);
+  *size = utf8_len - 1; /* utf8_len includes the null terminator */
+  buffer[*size] = '\0';
+
+  return 0;
+}
```

Comment:
I _think_ the Windows implementation ought to return UV_EINVAL and UV_ENOMEM but I defer to @piscisaureus on this one.